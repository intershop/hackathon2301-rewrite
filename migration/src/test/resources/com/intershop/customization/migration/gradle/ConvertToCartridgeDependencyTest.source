plugins {
    id 'com.intershop.gradle.cartridge-resourcelist'
    id 'com.intershop.gradle.isml'
    id 'com.intershop.icm.cartridge.external'
    id 'com.intershop.icm.cartridge.product'
    id 'java'
}

description = 'Application - Responsive Starter Store'


task lessCompile {
    ext.srcFile = file('staticfiles/cartridge/static/default/less/theme.less')
    ext.destFile = file('staticfiles/cartridge/static/default/css/theme.css')
    doLast {
        destFile.parentFile.mkdirs()

        def lessCompiler = new org.lesscss.LessCompiler()
        lessCompiler.compress = true
        lessCompiler.compile(srcFile, destFile)
    }
}


tasks.compileJava.dependsOn(tasks.lessCompile)

clean.doFirst {
    delete file('staticfiles/cartridge/static/default/css/theme.css')
}

tasks.test.dependsOn(tasks.isml)

// TODO please validate that cartridges have "cartridge" as dependency declaration instead of "implementation".
dependencies {
    implementation 'com.intershop.business:ac_cxml'
    implementation 'com.intershop.business:ac_gtm_sf'
    implementation 'com.intershop.business:ac_giftcard_std'
    implementation 'com.intershop.business:ac_payment_std'
    implementation 'com.intershop.business:bc_abtest'
    implementation 'com.intershop.business:bc_addresscheck'
    implementation 'com.intershop.business:bc_basket'
    implementation 'com.intershop.business:bc_catalog'
    implementation 'com.intershop.business:bc_catalog_orm'
    implementation 'com.intershop.business:bc_customer'
    implementation 'com.intershop.business:bc_customer_orm'
    implementation 'com.intershop.business:bc_giftcard'
    implementation 'com.intershop.business:bc_image'
    implementation 'com.intershop.business:bc_marketing'
    implementation 'com.intershop.business:bc_mvc'
    implementation 'com.intershop.business:bc_order'
    implementation 'com.intershop.business:bc_orderprocess'
    implementation 'com.intershop.business:bc_payment'
    implementation 'com.intershop.business:bc_pricing'
    implementation 'com.intershop.business:bc_product'
    implementation 'com.intershop.business:bc_product_orm'
    implementation 'com.intershop.business:bc_product_pricing'
    implementation 'com.intershop.business:bc_productbinding'
    implementation 'com.intershop.business:bc_promotion'
    implementation 'com.intershop.business:bc_rating'
    implementation 'com.intershop.business:bc_requisition'
    implementation 'com.intershop.business:bc_search'
    implementation 'com.intershop.business:bc_shipping'
    implementation 'com.intershop.business:bc_tendering'
    implementation 'com.intershop.business:bc_warranty'
    implementation 'com.intershop.business:bc_wishlist'
    implementation 'com.intershop.business:bc_wishlist_orm'
    implementation 'com.intershop.business:bts'
    implementation 'com.intershop.business:sld_ch_b2c_base'
    implementation 'com.intershop.business:sld_ch_b2c_image'
    implementation 'com.intershop.business:sld_ch_sf_base'
    implementation 'com.intershop.business:sld_mcm'
    implementation 'com.intershop.business:sld_pdf'
    implementation 'com.intershop.business:sld_pmc'
    implementation 'com.intershop.business:sld_preview'
    implementation 'com.intershop.business:xcs'
    implementation 'com.intershop.business:ac_inventory_service'
    implementation 'com.intershop.business:bc_basket_service'
    implementation 'com.intershop.content:bc_pmc'
    implementation 'com.intershop.content:bc_preview'
    implementation 'com.intershop.platform:ac_captcha_recaptcha'
    implementation 'com.intershop.platform:ac_ruleengine_drools'
    implementation 'com.intershop.platform:app'
    implementation 'com.intershop.platform:bc_address'
    implementation 'com.intershop.platform:bc_application'
    implementation 'com.intershop.platform:bc_captcha'
    implementation 'com.intershop.platform:bc_foundation'
    implementation 'com.intershop.platform:bc_i18n'
    implementation 'com.intershop.platform:bc_mail'
    implementation 'com.intershop.platform:bc_organization'
    implementation 'com.intershop.platform:bc_region'
    implementation 'com.intershop.platform:bc_repository'
    implementation 'com.intershop.platform:bc_service'
    implementation 'com.intershop.platform:bc_user'
    implementation 'com.intershop.platform:bc_user_orm'
    implementation 'com.intershop.platform:businessobject'
    implementation 'com.intershop.platform:cache'
    implementation 'com.intershop.platform:component'
    implementation 'com.intershop.platform:configuration'
    implementation 'com.intershop.platform:core'
    implementation 'com.intershop.platform:emf'
    implementation 'com.intershop.platform:isml'
    implementation 'com.intershop.platform:orm'
    implementation 'com.intershop.platform:pf_extension'
    implementation 'com.intershop.platform:pf_property'
    implementation 'com.intershop.platform:pipeline'
    implementation 'com.intershop.platform:servletengine'
    implementation 'com.intershop.platform:ui_web_library'
    implementation 'com.intershop.platform:rest'
    implementation 'com.intershop:api_service'

    implementation 'commons-collections:commons-collections'
    implementation 'commons-lang:commons-lang'
    implementation 'com.google.guava:guava'

    implementation 'org.apache.commons:commons-lang3'

    implementation 'org.eclipse.emf:org.eclipse.emf.common'
    implementation 'org.eclipse.emf:org.eclipse.emf.ecore'

    runtimeOnly 'com.intershop.business:app_sf_rest_b2c'

    testImplementation 'org.mockito:mockito-core'
    testImplementation 'org.hamcrest:hamcrest-core'
    testImplementation 'org.hamcrest:hamcrest-library'
    testImplementation 'junit:junit'
    testImplementation 'org.junit.jupiter:junit-jupiter'
    testImplementation 'org.junit.jupiter:junit-jupiter-api'
    testImplementation 'org.junit.vintage:junit-vintage-engine'
    testImplementation 'org.junit.platform:junit-platform-runner'
    testImplementation 'com.intershop.platform:pf_core_test'

}
