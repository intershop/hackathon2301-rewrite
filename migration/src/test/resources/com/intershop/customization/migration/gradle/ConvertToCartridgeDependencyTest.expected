plugins {
    id 'com.intershop.gradle.cartridge-resourcelist'
    id 'com.intershop.gradle.isml'
    id 'com.intershop.icm.cartridge.external'
    id 'com.intershop.icm.cartridge.product'
    id 'java'
}

description = 'Application - Responsive Starter Store'


task lessCompile {
    ext.srcFile = file('staticfiles/cartridge/static/default/less/theme.less')
    ext.destFile = file('staticfiles/cartridge/static/default/css/theme.css')
    doLast {
        destFile.parentFile.mkdirs()

        def lessCompiler = new org.lesscss.LessCompiler()
        lessCompiler.compress = true
        lessCompiler.compile(srcFile, destFile)
    }
}


tasks.compileJava.dependsOn(tasks.lessCompile)

clean.doFirst {
    delete file('staticfiles/cartridge/static/default/css/theme.css')
}

tasks.test.dependsOn(tasks.isml)

// TODO please validate that cartridges have "cartridge" as dependency declaration instead of "implementation".
dependencies {
    cartridge 'com.intershop.business:ac_cxml'
    cartridge 'com.intershop.business:ac_gtm_sf'
    cartridge 'com.intershop.business:ac_giftcard_std'
    cartridge 'com.intershop.business:ac_payment_std'
    cartridge 'com.intershop.business:bc_abtest'
    cartridge 'com.intershop.business:bc_addresscheck'
    cartridge 'com.intershop.business:bc_basket'
    cartridge 'com.intershop.business:bc_catalog'
    cartridge 'com.intershop.business:bc_catalog_orm'
    cartridge 'com.intershop.business:bc_customer'
    cartridge 'com.intershop.business:bc_customer_orm'
    cartridge 'com.intershop.business:bc_giftcard'
    cartridge 'com.intershop.business:bc_image'
    cartridge 'com.intershop.business:bc_marketing'
    cartridge 'com.intershop.business:bc_mvc'
    cartridge 'com.intershop.business:bc_order'
    cartridge 'com.intershop.business:bc_orderprocess'
    cartridge 'com.intershop.business:bc_payment'
    cartridge 'com.intershop.business:bc_pricing'
    cartridge 'com.intershop.business:bc_product'
    cartridge 'com.intershop.business:bc_product_orm'
    cartridge 'com.intershop.business:bc_product_pricing'
    cartridge 'com.intershop.business:bc_productbinding'
    cartridge 'com.intershop.business:bc_promotion'
    cartridge 'com.intershop.business:bc_rating'
    cartridge 'com.intershop.business:bc_requisition'
    cartridge 'com.intershop.business:bc_search'
    cartridge 'com.intershop.business:bc_shipping'
    cartridge 'com.intershop.business:bc_tendering'
    cartridge 'com.intershop.business:bc_warranty'
    cartridge 'com.intershop.business:bc_wishlist'
    cartridge 'com.intershop.business:bc_wishlist_orm'
    cartridge 'com.intershop.business:bts'
    cartridge 'com.intershop.business:sld_ch_b2c_base'
    cartridge 'com.intershop.business:sld_ch_b2c_image'
    cartridge 'com.intershop.business:sld_ch_sf_base'
    cartridge 'com.intershop.business:sld_mcm'
    cartridge 'com.intershop.business:sld_pdf'
    cartridge 'com.intershop.business:sld_pmc'
    cartridge 'com.intershop.business:sld_preview'
    cartridge 'com.intershop.business:xcs'
    cartridge 'com.intershop.business:ac_inventory_service'
    cartridge 'com.intershop.business:bc_basket_service'
    cartridge 'com.intershop.content:bc_pmc'
    cartridge 'com.intershop.content:bc_preview'
    cartridge 'com.intershop.platform:ac_captcha_recaptcha'
    cartridge 'com.intershop.platform:ac_ruleengine_drools'
    cartridge 'com.intershop.platform:app'
    cartridge 'com.intershop.platform:bc_address'
    cartridge 'com.intershop.platform:bc_application'
    cartridge 'com.intershop.platform:bc_captcha'
    cartridge 'com.intershop.platform:bc_foundation'
    cartridge 'com.intershop.platform:bc_i18n'
    cartridge 'com.intershop.platform:bc_mail'
    cartridge 'com.intershop.platform:bc_organization'
    cartridge 'com.intershop.platform:bc_region'
    cartridge 'com.intershop.platform:bc_repository'
    cartridge 'com.intershop.platform:bc_service'
    cartridge 'com.intershop.platform:bc_user'
    cartridge 'com.intershop.platform:bc_user_orm'
    cartridge 'com.intershop.platform:businessobject'
    cartridge 'com.intershop.platform:cache'
    cartridge 'com.intershop.platform:component'
    cartridge 'com.intershop.platform:configuration'
    cartridge 'com.intershop.platform:core'
    cartridge 'com.intershop.platform:emf'
    cartridge 'com.intershop.platform:isml'
    cartridge 'com.intershop.platform:orm'
    cartridge 'com.intershop.platform:pf_extension'
    cartridge 'com.intershop.platform:pf_property'
    cartridge 'com.intershop.platform:pipeline'
    cartridge 'com.intershop.platform:servletengine'
    cartridge 'com.intershop.platform:ui_web_library'
    cartridge 'com.intershop.platform:rest'
    implementation 'com.intershop:api_service'

    implementation 'commons-collections:commons-collections'
    implementation 'commons-lang:commons-lang'
    implementation 'com.google.guava:guava'

    implementation 'org.apache.commons:commons-lang3'

    implementation 'org.eclipse.emf:org.eclipse.emf.common'
    implementation 'org.eclipse.emf:org.eclipse.emf.ecore'

    cartridgeRuntime 'com.intershop.business:app_sf_rest_b2c'

    testImplementation 'org.mockito:mockito-core'
    testImplementation 'org.hamcrest:hamcrest-core'
    testImplementation 'org.hamcrest:hamcrest-library'
    testImplementation 'junit:junit'
    testImplementation 'org.junit.jupiter:junit-jupiter'
    testImplementation 'org.junit.jupiter:junit-jupiter-api'
    testImplementation 'org.junit.vintage:junit-vintage-engine'
    testImplementation 'org.junit.platform:junit-platform-runner'
    testImplementation 'com.intershop.platform:pf_core_test'

}
